packages/2d/src/components/Knot.ts

```diff
@@ -0,0 +1,183 @@
+   * auto-calculated position by using the {@link auto} property.
+   *
+   * @defaultValue Mirrored position of the endHandle.
+   */
+  @initial(() => ((<unknown>this) as Knot).getDefaultStartHandle)
+  @wrapper(Vector2)
+  @signal()
+  public declare readonly startHandle: Vector2Signal<this>;
+
+  /**
+   * The position of the knot's end handle. The position is provided relative
+   * to the knot's position.
+   *
+   * @remarks
+   * By default, the position of the end handle will be the mirrored position
+   * of the {@link startHandle}.
+   *
+   * If neither an end handle nor a start handle is provided, the positions of
+   * the handles gets calculated automatically to create smooth curve through
+   * the knot. The smoothness of the resulting curve can be controlled via the
+   * {@link Spline.smoothness} property.
+   *
+   * It is also possible to blend between a user-defined position and the
+   * auto-calculated position by using the {@link auto} property.
+   *
+   * @defaultValue Mirrored position of the startHandle.
+   */
+  @initial(() => ((<unknown>this) as Knot).getDefaultEndHandle)
+  @wrapper(Vector2)
+  @signal()
+  public declare readonly endHandle: Vector2Signal<this>;
+
+  /**
+   * How much to blend between the user-provided handles and the auto-calculated
+   * handles.
+   *
+   * @remarks
+   * This property has no effect if no explicit handles are provided for the
+   * knot.
+   *
+   * @defaultValue 0
+   */
+  @cloneable(false)
+  @initial(() => ({startHandle: 0, endHandle: 0}))
+  @parser((value: PossibleKnotAuto) => {
+    if (typeof value === 'object' && !Array.isArray(value)) {
+      return value;
+    }
+    if (typeof value === 'number') {
+      value = [value, value];
+    }
+    return {startHandle: value[0], endHandle: value[1]};
+  })
+  @compound({startHandle: 'startHandleAuto', endHandle: 'endHandleAuto'})
+  public declare readonly auto: KnotAutoSignal<this>;
+  public get startHandleAuto() {
+    return this.auto.startHandle;
+  }
+  public get endHandleAuto() {
+    return this.auto.endHandle;
+  }
+
+  public constructor(props: KnotProps) {
+    super(props);
+  }
+
+  @computed()
+  public points(): SerializedKnot {
+    const hasStartHandle = !this.startHandle.isInitial();
+    const hasEndHandle = !this.endHandle.isInitial();
+
+    // If no explicit handles are provided, the knot should use the
+    // auto-calculated handles.
+    if (!hasStartHandle && !hasEndHandle) {
+      const hasStartHandleAuto = !this.startHandleAuto.isInitial();
+      const hasEndHandleAuto = !this.endHandleAuto.isInitial();
+
+      // If an `auto` value was provided, we interpret this to mean that the
+      // user wants to blend between the auto-calculated handles and a handle
+      // that sits at the origin of the knot.
+      // If no `auto` value is provided either, we simply set the `auto` value
+      // to 1 to use position of the auto-calculated handles.
```

Comment:
Wouldn't that mean that the user would always have to explicitly pass the `auto` parameter? This was meant as a shortcut for `auto = 1` when no handles are provided. 

We could just ignore the `auto` parameter if no handles are provided though, instead of implicitly treating the handles as `0` in that case.

Reply:
>This was meant as a shortcut for auto = 1 when no handles are provided.

Maybe a better idea would be to only modify the property during initialization?
Something like:
```tsx
public constructor(props: KnotProps) {
  super(
    props.startHandle === undefined && props.endHandle === undefined
      ? {auto: 1, ...props}
      : props,
  );
}
```

Reply:
Yeah that seems to work fine. It also simplifies the `points` method quite a bit 👍